Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -313,6 +321,41 @@ private class ParameterSetNames
         [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoCopy)]
         public SwitchParameter Copy { get; set; }
 
+        /// <summary>
+        /// <para type="description">
+        /// The source table to copy from.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.DoCopy)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Table.TableReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(TableList.TablesData),
+            Property = nameof(TableList.TablesData.TableReference))]
+        [ValidateNotNull]
+        public TableReference Source { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The destination table to write to.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.DoCopy)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Table.TableReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(TableList.TablesData),
+            Property = nameof(TableList.TablesData.TableReference))]
+        [ValidateNotNull]
+        public TableReference Destination { get; set; }
```

Comment:
This parameter and `DestinationTable` can be one parameter. Just add a second `[Parameter()]` attribute for the new parameter set.